### ISSUE TITLE
Enrich documentation

### ISSUE THREAD

**Issue by uruha** [uruha/next-with-ts#55]
## Loadmap of enrich documentation for this repository
- [x] redux/middleware architecture image
- [x] testing guidance
  - [x] component testing
  - [x] redux testing
  - [x] other origin lib/util testing
- [x] file tree description
- ~~security policy~~

**Comment by uruha** [comment]
Working

```
## Test policy
### Component
Mainly use testing-library/react, react-hooks.

#### Checkpoints
1. After fire event, compare target state or props differences
2. With or without props, examine the state of the output HTMLElement contained by Component

### Redux / Redux-saga
#### Action [priority: *LOW*]
Normaly simple unit testing.
Because of actions is pure function and have types for I/O,
If types is safe, there is without having to do little testing.

#### Reducer [priority: *MIDDLE*]
Reducer testing have two types, unit and snapshot testing.
Unit testing is compare initial state to updated state.
Snapshot testing is check the difference visually for target state.

##### Why do snapshot testing ?
[要英訳]
reducer は任意の action に対して、定義した reducer のスコープ上の state (object) を更新する。
すなわち、action のバリエーションだけ state の変化を確認する必要があり、値を比較することでカバーするのはあまり現実的では無い。
そこで、任意の action に対してリグレッションテストを行うことを提案する。
リグレッションであれば、pull request で常に state の状態を目視で確認することができる。
仕様が変われば、PR でその変更点をチーム全員でディスカッションができるだろう。
リグレッションテストに値を考慮する必要がない。
reducer がそのコードによってどのような変化を state に与えるかを確認するだけだ。

The reducer updates the state (object) on the defined reducer scope for any action.
In other words, it is necessary to check the change of state only for the variation of action, and it is not realistic to cover by comparing the values.
Therefore, we propose to perform a regression test for any action.
If it is a regression, you can always check the state of the state visually with a pull request.
If the specs change, the team will be able to discuss the changes in the PR.
No need to consider values in regression tests.
Just see what the reducer changes the state with the code.

#### Redux-saga [priority: *HIGH*]
[WIP]

### Other
[WIP]
```
